### PR TITLE
[BAQE-439] Introducing Narayana and tomcat-dbcp into engine tests

### PR DESCRIPTION
**Pull request by rsynek**
@mswiderski @MarianMacik sorry for such a big PR, I reworked the PoolingDataSource API that was spread across the entire code base.

The PR depends on https://github.com/kiegroup/drools/pull/1939.

### CODE REVIEW

**Comment by rsynek** [comment]
jenkins retest this

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/HumanTaskServicesBaseTest.java [hunk 1]
```diff
@@ -185,10 +185,10 @@ protected JaxbContent xmlRoundTripContent(Content content) {
     protected static final String PASSWORD = "password";
     protected static final String JDBC_URL = "url";
 
-    protected static PoolingDataSource setupPoolingDataSource() {
-        Properties dsProps = getDatasourceProperties();
-        PoolingDataSource pds = PersistenceUtil.setupPoolingDataSource(dsProps, "jdbc/jbpm-ds", false);
-        pds.init();
+    protected static PoolingDataSourceWrapper setupPoolingDataSource() {
+        //Properties dsProps = getDatasourceProperties();
```
Comment (mswiderski) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/HumanTaskServicesBaseTest.java]: remove the commented code if not needed

**Comment by rsynek** [comment]
@mswiderski good question, there are other utilities in the package, but probably not all the modules use them. Will find out where the dependency can be removed.

**Comment by mswiderski** [comment]
@rsynek would be worth checking if we can move them to kie-test-util and thus get rid of one module

**Comment by rsynek** [comment]
@mswiderski there are classes closely related to testing processes (e.g. various listeners); so this doesn't look like an option - there is nothing related to Drools.

**Comment by mswiderski** [comment]
@rsynek ok, let's keep it as is, thanks let me know when this is ready to be merged

**Comment by rsynek** [comment]
@mswiderski the jbpm-test-util is still required by other modules due to the classes I mentioned. Let's merge.

**Pull request merged**